testsuite/tests/lib-string/test_string.ml

```diff
@@ -76,3 +76,59 @@ let () =
   assert(not (String.ends_with ~suffix:"foo" ""));
   assert(not (String.ends_with ~suffix:"obaz" "foobar"));
 ;;
+
+
+let () =
+  let test ?limit x y d =
+    assert (String.edit_distance ?limit x y = d);
+    assert (String.edit_distance ?limit y x = d);
+    assert (String.edit_distance ?limit x x = 0);
+    assert (String.edit_distance ?limit y y = 0);
+  in
+  test "" "" 0;
+  test "" "ab" 2;
+  test "function" "function" 0;
+  test "function" "fanction" 1;  (* substitute *)
+  test "function" "fnction" 1;   (* delete *)
+  test "function" "funiction" 1; (* insert *)
+  test "function" "funtcion" 1;  (* transpose *)
+  test "function" "fantcion" 2;  (* substitute + transpose *)
+  test "function" "fantcio" 3;   (* substitute + transpose + delete *)
+  test "function" "efantcio" 4;  (* all *)
+  test "fun" "function" 5;
+  test "fun" "function" ~limit:0 0;
+  test "fun" "function" ~limit:1 1;
+  test "fun" "function" ~limit:2 2;
+  test "fun" "function" ~limit:3 3;
+  test "fun" "function" ~limit:4 4;
+  test "fun" "function" ~limit:5 5;
+  test "fun" "function" ~limit:6 5;
+  test "ca" "abc" 3 (* Damerau-Levenshtein would be 2 *);
+  test "élément" "élment" 1;
+  test "OCaml🐫" "O'Caml🐪" 2;
+;;
+
+let () =
+  let test ?max_dist dict s res =
+    let dict = fun yield -> List.iter yield dict in
+    assert (String.spellcheck ?max_dist dict s = res)
+  in
+  (* max_dist = 0 *)
+  test [""] "" [""];
+  test ["a"; "b"] "" [];
+  test ["a"; "b"] "a" ["a"];
+  test ["a"; "b"] "d" [];
+  test ["a"; "b"] "é" [];
+  test ["aa"; "aé"] "aé" ["aé"];
+  test ["aa"; "aé"] "ad" [];
+  (* max_dist = 1 *)
+  test ["abc"; "abcé"] "abc" ["abc"];
+  test ["abc"; "abcé"; "abcéd"] "abé" ["abc"; "abcé"];
+  test ["abcdé"; "abcdéf"] "abcd" ["abcdé"];
+  (* max_dist = 2 *)
+  test ["abcdéf"] "abcde" ["abcdéf"];
+  test ["abcdéf"] "ubcde" [];
+  let max_dist s = if String.length s <= 1 then 1 else 2 in
+  test ~max_dist ["abc"] "a" [];
+  test ~max_dist ["abc"; "ab"; "b"] "a" ["ab"; "b"];
+  ()
```

Comment:
This file lacks a `non-ascii` attribute in https://github.com/ocaml/ocaml/blob/trunk/.gitattributes .